### PR TITLE
Cleanup changes : Details in the comment

### PR DESCRIPTION
**Pull request by subhankarc**
* Handling failures in the server.js and exiting in case of failures
* Removing lib.bootstrap call from server.js and also removing it from broker/lib/index.js
* Instead, moving the docker bootstraping call in the start of the docker operator.
* Moving loadCatalogFromAPIServer from broker/lib/index.js to utils
* Removing load of BOSHDirectorClient and Bosh cache in Plan, Changing the get stemcell method
* Removing unnecessary object cloning of catalog for CfPlatformmanager

### CODE REVIEW

File: broker/lib/index.js [hunk 1]
```diff
@@ -1,45 +1,4 @@
...
-            });
-        }));
-      })
-      .then(() => catalog.reload())
-      .tap(() => logger.silly('Loaded Services in catalog Are ', catalog.services))
-      .tap(() => logger.silly('Loaded Plans in catalog Are ', catalog.plans));
-  }
-}
\ No newline at end of file
+exports.middleware = require('./middleware');
```
[the remark this answers is not in the document]
Comment (amitmalav) [reply broker/lib/index.js]: Okay. We can do it as part of other PR

File: broker/server.js [hunk 2]
```diff
@@ -9,55 +9,56 @@ const HttpServer = require('../common/HttpServer');
...
-      app.use('/manage', routes.manage);
-    });
-    HttpServer.start(external);
+      HttpServer.start(external);
+    }
+    HttpServer.handleShutdown(); //https://github.com/nodejs/node-v0.x-archive/issues/5054
+    require('../common/UnlockResourcePoller');
+  } catch (error) {
+    logger.info('Process shutting down now because of error ', error);
+    process.exit(2);
```
[the remark this answers is not in the document]
Comment (amitmalav) [reply broker/server.js]: https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/master/common/constants.js#L230

File: broker/lib/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (subhankarc) [reply broker/lib/index.js]: sure! one at a time!

File: broker/server.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (subhankarc) [reply broker/server.js]: done

**Pull request merged**